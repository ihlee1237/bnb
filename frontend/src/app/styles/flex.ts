import styled from "styled-components";

export const Flex = styled.div<{ gap?: number }>`
  display: flex;
  gap: ${({ gap }) => (gap ? `${gap}px` : "0")};
`;

export const FlexRow = styled(Flex)`
  flex-direction: row;
`;

export const FlexColumn = styled(Flex)`
  flex-direction: column;
`;
