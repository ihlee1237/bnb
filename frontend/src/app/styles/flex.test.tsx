import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Flex, FlexColumn, FlexRow } from "./flex";

describe("Flex", () => {
  it("Flex snapshot", () => {
    const { container } = render(<Flex $gap={10} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Flex with no gap", () => {
    const { container } = render(<Flex />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("FlexRow snapshot", () => {
    const { container } = render(<FlexRow />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("FlexColumn", () => {
    const { container } = render(<FlexColumn />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
