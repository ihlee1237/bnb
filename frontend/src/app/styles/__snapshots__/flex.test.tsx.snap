// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Flex > Flex snapshot 1`] = `
<div
  class="sc-bXdtCk hQcNhb"
  gap="10"
/>
`;

exports[`Flex > Flex with no gap 1`] = `
<div
  class="sc-bXdtCk kBBVNO"
/>
`;

exports[`Flex > FlexColumn 1`] = `
<div
  class="sc-bXdtCk sc-fmZSGO kBBVNO ftTnuu"
/>
`;

exports[`Flex > FlexRow snapshot 1`] = `
<div
  class="sc-bXdtCk sc-hKDTPf kBBVNO KflQV"
/>
`;
